sourcegraph/sourcegraph
# Cody repositories selector design cleanups

This fixes a few layout bugs and cleans up some of the design details on the new repository selector popover.

* Better visibility for disabled inferred items
* Hover styles for clickable rows
* Improve the embeddings status text (as per https://github.com/sourcegraph/sourcegraph/pull/53165#issuecomment-1583677432 but without the link to docs, as I couldn't figure that out). It's now consistent in behaviour and text between both the popover and on the icon in the compose window. It did mean switching to a plain tooltip, but I think that's okay for now until we figure out how we want to talk about "quality" (if we want to do that at all)
* Doesn't show the Reset button unless there's something to reset
* Switches to a `type="search"` input for the search (escape key behaviour could still be improved to only bubble up to the popover only if the search is empty)
* Fixes position of button content if labels are long and truncated
* Adjusts the text more dependently for inferred-less pages (Cody top level) vs contextual pages
* Adds an icon to the empty state on the top-level Cody page
* Removes the (x) button when doing a search, as it can be confused with clearing the search
* Fixing the gradient "all embeddings are okay" version of the button so it's the same size and doesn't shift

### Before

https://github.com/sourcegraph/sourcegraph/assets/153/cff7dd95-ccbe-4bc7-8c91-138478c886f4 

https://github.com/sourcegraph/sourcegraph/assets/153/4d11b8c2-e2ca-4ec3-b5b9-8a3d290a8e44

<img width="454" alt="Screenshot 2023-06-09 at 5 15 36 pm" src="https://github.com/sourcegraph/sourcegraph/assets/153/9cacee3a-00ff-415c-90c2-1e215f2909cb">

### After

https://github.com/sourcegraph/sourcegraph/assets/153/064497ab-11a3-46c8-b5c7-a25e3bff735c

https://github.com/sourcegraph/sourcegraph/assets/153/552b017b-bd6a-4758-9cb2-a2e3c83f532a

<img width="427" alt="Screenshot 2023-06-09 at 5 15 50 pm" src="https://github.com/sourcegraph/sourcegraph/assets/153/c07aeec8-d705-46c2-b61d-7a920ca1fc9b">

## Test plan

* Navigate to Cody AI and repo pages
* Use the selector with as many states
* Very all states display as expected
